[docs] Apply the new DX to the Button Field demos (#14860)
mui/mui-x: docs/data/date-pickers/custom-field/custom-field.md

## With a custom editing experience

### Using an Autocomplete

If your user can only select a value in a small list of available dates, you can replace the field with the [Autocomplete](/material-ui/react-autocomplete/) component to list those dates:

{{"demo": "PickerWithAutocompleteField.js", "defaultCodeOpen": false}}

### Using a read-only Text Field

If you want users to select a value exclusively through the views

> but you still want the UI to look like a Text Field, you can replace the field with a read-only [TextField](/material-ui/react-text-field/) component:

Reviewer: ```suggestion
but you still want the UI to look like a Text Field, you can replace the field with a read-only [Text Field](/material-ui/react-text-field/) component:
```

## With a custom editing experience

### Using an Autocomplete

If your user can only select a value in a small list of available dates,

> you can replace the field with the `<Autocomplete />` from `@mui/material` to those dates:

Reviewer 2: ```suggestion
you can replace the field with the `<Autocomplete />` from `@mui/material` to list those dates:
```

> you can replace the field with the `<Autocomplete />` from `@mui/material` to list those dates:

Reviewer: I would style this as a proper noun rather than two backticked elements.

```suggestion
you can replace the field with Material UI Autocomplete to list those dates:
```

Reviewer: I'm curious if @vladmoroz has any thoughts about formatting in this context.

Reviewer 3: I'd say it like this personally...

> If your user can only select a value in a small list of available dates, you can replace the field with [Autocomplete](https://mui.com/material-ui/react-autocomplete/) to list those dates:

This sounds OK to me too:

> If your user can only select a value in a small list of available dates, you can replace the field with the Autocomplete component from `@mui/material` to list those dates:

- Proper noun either way
- I'd link out to Autocomplete because I need the API handy, and it saves us a reference to Material UI. Otherwise, my preference is explicit `@mui/material` instead of "Material UI" because people confuse MUI / Material UI / Material Design all the time.
- Either "with the Autocomplete component from..." or "with Autocomplete from...". Just "the Autocomplete" doesn't sound right to my ear.

Author: I updated the 3 sentences with a link to our `@mui/material` doc.

I just kept the "with the [Autocomplete]() component" because just "with [Autocomplete()" felt weird to me (same for "with [Button]().".

{{"demo": "PickerWithAutocompleteField.js", "defaultCodeOpen": false}}

> ### Using a read-only Text Field

Reviewer 2: Not sure if this was intentional, but did you mean to use 'Text Field' or 'TextField' (like in the text below)?

```suggestion
### Using a read-only TextField
```

Reviewer 2: or according to this doc: https://www.notion.so/mui-org/Style-conventions-for-MUI-components-and-products-dff7d2d7d1ba4a4abd47f48cf345b800?pvs=4#1279b9e04d514d32a109b846a0945813

shouldn't we use 'text field' instead?

Author: We are describing the `<TextField />` from `@mui/material` here, not the generic concept of a text field, so I don't think "text field" would be appropriated.

I'm using "Text Field" because it's a reference to one our our product (from the core team), similar to the Data Grid or the Date Time Picker.

I'll ping Sam for a validation of this PR before merging :+1:

Reviewer 2: Well, technically it is not a product ... I guess... no? 🤔

Reviewer: We're describing our specific implementation (rather than the general concept), so I would style it as a proper noun: Text Field.

If you want users to select a value exclusively through the views

> but you still want the UI to look like a TextField, you can replace the field with a read-only `<TextField />` from `@mui/material`:

Reviewer: ```suggestion
but you still want the UI to look like a Text Field, you can replace the field with a read-only Material UI Text Field component:
```

Reviewer: I would also like to see the line break removed from this sentence—try to keep it to one sentence per line, even if it's a very long sentence.

Author: What is the goal of removing line break on the markdown file?
Those do not impact the line break on the doc..

Reviewer: It's just simpler to maintain one sentence per line—if I wanted to edit this sentence more heavily I'd have to leave two separate comments, or leave a suggestion across the two lines. We don't need to add our own line wrapping when an IDE will do that for us.

Author: > We don't need to add our own line wrapping when an IDE will do that for us.

Webstorm advises me to add those line breaks though :laughing:

Reviewer: Interesting! I don't think we should listen to Webstorm in this case. We don't have a consistent rule to fall back on to decide when to add a line break like this, so I think it would make the docs look really messy if we started doing this everywhere. If we wanted to implement a style rule like this then I think it would need to happen programmatically via Vale or Prettier maybe. I've never used Webstorm before, though. Does it wrap lines for you, or does it display a sentence like this as one long line?

Author: > it would make the docs look really messy if we started doing this everywhere

I'm basically doing it everywhere
But if we say that we don't do line break inside a sentence, that's fine for me.

> Does it wrap lines for you, or does it display a sentence like this as one long line?

It just adds a visual feedback, similar to the one when I have a spelling error, I can probably disable it.

Reviewer: > But if we say that we don't do line break inside a sentence, that's fine for me.

Let's stick with that then, for the sake of consistency between authors and products.

{{"demo": "behavior-read-only-text-field/MaterialDatePicker.js", "defaultCodeOpen": false}}

### Using a Button

If you want users to select a value exclusively through the views

> and you don't want the UI to look like a Text Field, you can replace the field with a `<Button />` from `@mui/material`:

Reviewer 2: same here

```suggestion
and you don't want the UI to look like a TextField, you can replace the field with a `<Button />` from `@mui/material`:
```

> and you don't want the UI to look like a Text Field, you can replace the field with a `<Button />` from `@mui/material`:

Reviewer: Here's how I would style this:

```suggestion
and you don't want the UI to look like a Text Field, you can replace the field with a Material UI Button:
```

{{"demo": "behavior-button/MaterialDatePicker.js", "defaultCodeOpen": false}}

> The same logic can be applied to any range picker:

Reviewer: let's capitalize this since we're referring to the MUI X components.

```suggestion
The same logic can be applied to any Range Picker:
```
